docs/semantics.md

```diff
@@ -95,7 +95,14 @@ The negative result for ```foo/``` will be cached, but that only helps with the
 To alleviate this, Cloud Storage FUSE supports a "type cache" on directory inodes. When type cache is enabled, each directory inode will maintain a mapping from the name of its children to whether those children are known to be files or directories or both. When a child is looked up, if the parent's cache says that the child is a file but not a directory, only one Cloud Storage object will need to be stated. Similarly if the child is a directory but not a file.
 
 The behavior of type cache is controlled by the following flags/config parameters:
-1.  Type-cache TTL: It controls the duration for which Cloud Storage FUSE allows the kernel to inode type attributes. It can be set in one of the following two ways.
+1. Type-cache capacity: This is configurable at per-directory level by setting `metadata-cache: type-cache-max-size-mb-per-dir` in config-file. This is the maximum size of type-cache per-directory in MiBs. By default, this is set at 16, which roughly equates to about a million entries. If you have folders containing more than a million items (folders or files) you may want to increase this, otherwise the caching will not function properly when listing that folder's contents:
```

Comment:
1. It should be very rare to have 1M "files" in a single "directory".
2. The type-cache capacity does not limit overall memory usage, as that will depend on the number of directories.

So for the vast majority of use cases, this change does nothing. Why are we adding code that will never be used?

Reply:
Discussed this offline with @bjornleffler and resolved this issue.